src/genie/libs/parser/iosxe/show_gnxi.py

```diff
@@ -0,0 +1,288 @@
+"""show_gnxi.py
+
+IOSXE parsers for the following commands
+    * 'show gnxi state'
+    * 'show gnxi state detail'
+"""
+
+from collections import defaultdict
+import re
+from enum import Enum
+
+from genie.metaparser import MetaParser
+
+
+def _enabled_disabled_to_bool(value : str):
+    ''' Returns True if argument string is "Enabled"
+        else False
+    '''
+    return value == "Enabled"
+
+def _up_down_to_bool(value : str):
+    ''' Returns True if argument string is "Up"
+        else False
+    '''
+    return value == "Up"
+
+
+class ShowGnxiStateSchema(MetaParser):
+    ''' Schema for:
+            show gnxi state
+    '''
+    schema = {
+            "oper_state": {
+                "admin_enabled": bool,
+                "oper_up": bool
+                }
+            }
+
+
+class ShowGnxiState(ShowGnxiStateSchema):
+    ''' Parser for:
+            show gnxi state
+    '''
+    cli_command = "show gnxi state"
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        ret_dict = {}
+
+        p_full = re.compile(r"(?P<admin_state>Disabled|Enabled)\s*(?P<oper_status>Down|Up)")
+
+        matches = p_full.search(out)
+
+        if matches:
+            groups = matches.groupdict()
+
+            ret_dict = { "oper_state": {} }
+
+            ret_dict["oper_state"]["admin_enabled"] = _enabled_disabled_to_bool(groups["admin_state"])
+            ret_dict["oper_state"]["oper_up"] = _up_down_to_bool(groups["oper_status"])
```

Comment:
Are you talking about the keys in `groups` or the keys in `ret_dict`? For the former, the output from the command doesn't vary. It will either display state and status or the process is not running and no match will occur. Returning an empty dictionary.